### ISSUE TITLE
Release 1.10

### ISSUE THREAD

**Issue by esteve** [autowarefoundation/autoware_ai#399]

**Comment by hakuturu583** [comment]
Sorry for my late.
New entry autowarefoundation/autoware#1795

**Comment by esteve** [comment]
@hakuturu583 thanks for letting us know, but unfortunately it's too late for this release to include autowarefoundation/autoware#1795. The goal of this ticket is to list all the features and improvements that will go into a release before starting the release, not after. This helps us have a vision of what the priorities are for a given release/milestone and how efforts should be directed.

In any case, it looks like autowarefoundation/autoware#1795 is a good candidate for Autoware 1.11, we'll add it to the next release.

**Comment by kfunaoka** [comment]
Open source community should have a certain amount of discipline to work with a lot of people. I added a procedure into Contribution Rules to declare which version each issue is targeting.
https://github.com/CPFL/Autoware/wiki/Contribution-Rules#development-workflow

@hakuturu583 Sorry for the insufficient announcement. Also I'd like to encourage you to improve Autoware more and more.  For that, let's talk what you are working on within the community :)

@esteve I've also talked with @hakuturu583. Though we can postpone autowarefoundation/autoware#1795, simulator is a very good tool for autonomous driving, and it's worth for Autoware and users. Since autowarefoundation/autoware#1795 does not affect the essential part of Autoware, we might be able to decide on importing autowarefoundation/autoware#1795 to test Autoware functions by challengers. How do you think in this case?

@hakuturu583 Any other comments?

**Comment by kfunaoka** [comment]
Sorry, I have to withdraw autowarefoundation/autoware#1693 and autowarefoundation/autoware_ai#324.

**Comment by esteve** [comment]
@kfunaoka It was mostly because I think that we should be stating what we're going to work on for the current cycle, so we can anticipate and properly test everything before making a new release. That doesn't mean we can't squeeze in a new feature in the middle of the cycle, but they should have a rationale and give enough time to test it.

Unfortunately this wasn't the case with autowarefoundation/autoware#1795, it broke the build, doesn't have any tests at all and was merged in a rush.

autowarefoundation/autoware#1795 is a very valuable contribution and I'd really like it getting merged, but it'd be good to be a bit more disciplined with releases and plan a bit better, it'd help users anticipate any potentially incompatible changes before a release, for example.

**Comment by hakuturu583** [comment]
@esteve autowarefoundation/autoware#1795 have already merged to develop, and travis CI said that the develop branch was not broken.
Please show me the CI log.

**Comment by esteve** [comment]
@hakuturu583 no worries, it's really not big of a deal :-) I think the work you did in autowarefoundation/autoware#1795 is really valuable, but let's plan releases in advance so we can have more time to test them more thoroughly.

**Comment by kfunaoka** [comment]
@sgermanserrano @esteve Would you list pull requests that are better to be merged into v1.10? My understanding is:

- Use colcon as the build tool autowarefoundation/autoware#1704
- Switch to Apache 2 license (develop branch) autowarefoundation/autoware#1741
- feature/ndt_matching_disk_full autowarefoundation/autoware#1824

**Comment by sgermanserrano** [comment]
@kfunaoka @aohsato @cirpue49 as discussed in Slack #motion-planning channel `A*` is better used with `costmap_generator` than with `points2costmap` (which seems older -> can be deprecated and substituted by `costmap_generator`?). 
I'd suggest pushing [Feature] Improve Hybrid A* planner autowarefoundation/autoware#1594 from v1.10 to v1.11 to:
- Update from nav_msgs::OccupancyGrid to GridMap
- Review and improve tests: autowarefoundation/autoware#1753 
- Add tests for `freespace_planner` and `astar_avoid`

**Comment by esteve** [comment]
@kfunaoka I postponed autowarefoundation/autoware#1704 because I thought it'd make it for the original 1.10 cutoff, but since it was delayed, I think we can try. I'll keep testing it locally, I'd appreciate your (and @sgermanserrano  's) feedback, since it will have a direct impact on the users.

autowarefoundation/autoware#1741 should be mergeable, but I need another +1.

**Comment by kfunaoka** [comment]
I've updated the status.

- I've added autowarefoundation/autoware#1824 into the 1.10 list above since localization death is fatal for autonomous driving. It became a problem in our long duration test.
- autowarefoundation/autoware#1704 is almost ready to merge.
- autowarefoundation/autoware#1741 is lamost ready to merge but need a fix.

**Comment by esteve** [comment]
@kfunaoka I've added autowarefoundation/autoware#1704 to the list of tickets for this release and merged the PR.

**Comment by esteve** [comment]
Once autowarefoundation/autoware#1741 passes, I'll merge it as well.

**Comment by sgermanserrano** [comment]
@isamu-takagi @messi49 do autowarefoundation/autoware_ai#375 and autowarefoundation/autoware_ai#408 have associated PRs? or are they features on a planning/developing stage?

**Comment by kfunaoka** [comment]
@amc-nu @sgermanserrano @esteve How about creating release branch for v1.10 from HEAD of develop, merging autowarefoundation/autoware#1837 into the release branch, and staring test on the release branch?

- autowarefoundation/autoware#1837: Should be merged into v1.10.
- autowarefoundation/autoware_ai#408: Need more discussion. -> v1.11
- autowarefoundation/autoware#1645: Might take some time. -> v1.11?

**Comment by kfunaoka** [comment]
- autowarefoundation/autoware#1832: Can be merged into v1.10 to minimize time and effort for test.

**Comment by amc-nu** [comment]
Hello @esteve @sgermanserrano and @kfunaoka 

Please don't include autowarefoundation/autoware#1704 as it is in 1.10
I found the following problems:
1. `run` script won't work, as it is.
Please refer to https://github.com/CPFL/Autoware/blob/develop/ros/run#L52
2. Even after updating the script to `install` space. Since currently most of the packages **do not** include the `install` command for launch, and companion files. Only manual `ros run` commands will work, which are mostly not used by the UI and the current documentation.
https://github.com/CPFL/Autoware/blob/develop/ros/src/util/packages/runtime_manager/scripts/computing.yaml
3. Compiling with catkin_make workflow is also broken, so it cannot be used as an alternative.

Fixing the above points will require a considerable amount of work. For this reason, I consider appropriate 
1. First to apply autowarefoundation/autoware_ai#389 , then 
2. Fix the file installation commands in the CMakeFiles and 
3. Finally include autowarefoundation/autoware#1704 + fixes in a release.

Otherwise, current users (ourselves included) will face a lot of problems. Please let me know your thoughts on this.

**Comment by esteve** [comment]
@amc-nu I've reverted autowarefoundation/autoware#1704 in https://github.com/CPFL/Autoware/pull/1838, please have a look at it. However, the real problem is that for a long time the Autoware run script assumed that it was using the devel space, which is not a correct workflow in ROS. colcon is stricter about that, there's no devel space at all, only build and install, so if the packages are not properly written and they don't have an install target, they can't be used.

What I find strange is that catkin fails for you. Doesn't catkin_make_release work for you?

**Comment by kfunaoka** [comment]
I found the error autowarefoundation/autoware_ai#490. I'll check.
